### PR TITLE
Added find by date to EnvelopeRepository

### PR DESCRIPTION
**Pull request by apereverzin**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/BPS-1026

### Change description ###
Added 'find by date' to EnvelopeRepository. This 'find' method will be called from the service which will convert the report date into the datetime interval.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/blobrouter/data/EnvelopeRepository.java [hunk 1]
```diff
@@ -75,6 +76,16 @@ public EnvelopeRepository(NamedParameterJdbcTemplate jdbcTemplate, EnvelopeMappe
         );
     }
 
+    public List<Envelope> find(Instant from, Instant to) {
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/blobrouter/data/EnvelopeRepository.java]: maybe all we need is one parameter - date? (without the time)

**Comment by lgonczar** [comment]
Maybe we should prepare the report based on events. Otherwise, we won't include the files that we failed to processed. Worth having a discussion.

File: src/main/java/uk/gov/hmcts/reform/blobrouter/data/EnvelopeRepository.java (hunk 1 above)
Comment (apereverzin) [reply src/main/java/uk/gov/hmcts/reform/blobrouter/data/EnvelopeRepository.java]: One parameter - date - will be in a service method, this method will convert the date into the from - to range

Comment (rkondratowicz) [reply src/main/java/uk/gov/hmcts/reform/blobrouter/data/EnvelopeRepository.java]: if you think we'll need range somewhere else...

File: src/test/java/uk/gov/hmcts/reform/blobrouter/tasks/processors/BlobProcessorTest.java [hunk 2]
```diff
@@ -265,7 +266,7 @@ private void blobExists(OffsetDateTime time, byte[] contentToDownload) {
         }
 
         given(blobProperties.getLastModified()).willReturn(time);
-        given(envelopeRepo.find(any(), any())).willReturn(Optional.empty());
+        given(envelopeRepo.find(anyString(), anyString())).willReturn(Optional.empty());
```
Comment (rkondratowicz) [inline src/test/java/uk/gov/hmcts/reform/blobrouter/tasks/processors/BlobProcessorTest.java]: unrelated

Comment (apereverzin) [reply src/test/java/uk/gov/hmcts/reform/blobrouter/tasks/processors/BlobProcessorTest.java]: It is a fix for a compilation error :(

File: src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/data/EnvelopeRepositoryTest.java [hunk 3]
```diff
@@ -183,6 +189,32 @@ void should_return_empty_optional_when_envelope_for_given_container_and_file_nam
...
+        // given
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss");
+        Instant instant1 = LocalDateTime.parse("2019-12-19 10:31:25", formatter).toInstant(ZoneOffset.UTC);
+        Instant instant2 = LocalDateTime.parse("2019-12-20 11:32:26", formatter).toInstant(ZoneOffset.UTC);
+        Instant instant3 = LocalDateTime.parse("2019-12-20 12:33:27", formatter).toInstant(ZoneOffset.UTC);
+        Instant instant4 = LocalDateTime.parse("2019-12-21 13:34:28", formatter).toInstant(ZoneOffset.UTC);
+        repo.insert(new NewEnvelope("a", "b", instant1, instant1, Status.DISPATCHED));
+        repo.insert(new NewEnvelope("c", "d", instant2, instant2, Status.DISPATCHED));
+        repo.insert(new NewEnvelope("e", "f", instant3, instant3, Status.DISPATCHED));
+        repo.insert(new NewEnvelope("g", "h", instant4, instant4, Status.DISPATCHED));
```
Comment (rkondratowicz) [inline src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/data/EnvelopeRepositoryTest.java]: you've set the same instant in two fields of envelope, test name or test data don't explain which property of envelope is used for search

**Comment by rkondratowicz** [comment]
> Maybe we should prepare the report based on events. Otherwise, we won't include the files that we failed to process. Worth having a discussion.

@lgonczar Maybe we should introduce a new status and create an envelope?

File: src/test/java/uk/gov/hmcts/reform/blobrouter/tasks/processors/BlobProcessorTest.java (hunk 2 above)
Comment (rkondratowicz) [reply src/test/java/uk/gov/hmcts/reform/blobrouter/tasks/processors/BlobProcessorTest.java]: oh, you've added a new method in repo with signature similar to already existing one

**Comment by rkondratowicz** [comment]
Now that events repository exists we can use it instead

**Comment by rkondratowicz** [comment]
@apereverzin can we close this PR?
